### DISCUSSION TITLE
Localization

### DISCUSSION THREAD

**Discussion by Morganamilo** [Morganamilo/paru#433] [General]
Now that localization is implemented, all that's left is for people to write translations.

So I'm "closing" #424 as that's discussion about implementing localization in the code and here if for actually translating.

You can find translating instructions in [CONTRIBUTING.md](https://github.com/Morganamilo/paru/blob/master/CONTRIBUTING.md) as well as an exmple here https://github.com/Morganamilo/paru/commit/449d342fc0d35475fb454d73cd1f74151fddd3a6.

**Comment by ouuan** [comment]
`paru-git` is not localized yet. Do we need to install `.mo` files?

**Reply by Morganamilo** [reply]
Forgot to push. Now it's localized.

**Comment by mocdaniel** [comment]
I will gladly provide German localization later today.

**Comment by BachoSeven** [comment]
I'm working on the Italian translation, will open a PR soon.

**Reply by BachoSeven** [reply]
By the way, I found a typo [here](https://github.com/Morganamilo/paru/blob/master/src/install.rs#L698)(`.SRINFO` should be `.SRCINFO`), it's consequently also present in the `.pot` file and the currently present translations

(I've corrected it in the translation but non in the `msgid` ofc)

**Comment by Morganamilo** [comment]
Thanks I'll probably update that once the influx of translations stop.

**Reply by BachoSeven** [reply]
I found [this](https://github.com/BachoSeven/paru/blob/master/src/install.rs#L1670) as well, not sure if it's to be considered a typo though(`packagelist`-->`package list`) since maybe it's directly referencing the respective `makepkg` option(if this is the case, then I won't translate it)

**Reply by Morganamilo** [reply]
I guess it's debatable. The Chinese translation did translate it.

**Reply by BachoSeven** [reply]
Do you think it's important to keep the specific term? I think I'll translate it but I can put `(packagelist)` at the end..

**Reply by Morganamilo** [reply]
It's fine. packagelist is self explanatory and doesn't have directly relate to the makepkg flag. So just translate and don't worry about (packagelist).

**Comment by Paragoumba** [comment]
I'm working on the French translation

**Reply by michaelb** [reply]
I'll be able to review / proof-read that

**Reply by Silejonu** [reply]
Hi, how is the translation going? Have you started?
I'll have time to make a full translation tomorrow or the day after, but I don't want to duplicate work if you're already working on it.

**Reply by Paragoumba** [reply]
Hi, I translated all the strings but there might be some changes to do. I will commit everything so you can review it

**Reply by Paragoumba** [reply]
Pushed the changes to [my fork](https://github.com/Paragoumba/paru/commit/dbf986548ebb89f757a6dd8a1190761b676d29f9)

**Comment by Fifis** [comment]
Russian will be ready soon.

**Comment by is-hoku** [comment]
I'm working on the Japanese translation.

**Comment by wiredhikari** [comment]
I am working on hindi translation.

**Comment by rail01** [comment]
I am working on Polish i18n

**Reply by lukaszmoroz** [reply]
Are you still working on it? I've started working on it too before I saw this post, so I can help you finishing the work.

**Comment by orhun** [comment]
I'm working on the Turkish translation.

edit: saw #437, added review.

**Comment by ojmarcelino** [comment]
I'm working on (pt-br) Brazilian Portuguese, adding review on #437 right next to this.

**Comment by developomp** [comment]
Working on Korean translation

**Reply by developomp** [reply]
Stuff related to college got in my way so if anyone is interested in doing the work themself you may do so.

**Comment by stepanse** [comment]
I'll start working on Czech (cs) transaction.

**Comment by adriancostin6** [comment]
working on a translation to Romanian (ro). Will submit a PR once I'm done.

**Comment by botiapa** [comment]
Is anyone working on the Hungarian (hu) translation? If not, I'd gladly to it myself.

**Comment by Silejonu** [comment]
There are two strings that are very similar. Is there a reason for their difference or is this just a small mistake?

line 57

```
#: src/command_line.rs:158 src/command_line.rs:327
msgid "option {} does not allow a value"
msgstr ""
```

line 129

```
#: src/config.rs:896
msgid "option '{}' does not take a value"
msgstr ""
```

**Reply by developomp** [reply]
They are in two different places in the source code and need to get mapped accordingly

**Comment by Morganamilo** [comment]
I've now added localization for the --help output.

**Reply by flipflop97** [reply]
Hi, would it be possible to split these help lines in the translation into arguments and descriptions, and then dynamically produce the help text? Right now the whole line is translatable, which means the flag (eg `--makepkg`) can be translated while it shouldn't, and spacing must me manually changed too, which might produce inconsistent results within and between languages. Doing this will also reduce duplicates (many `<file>` or `<flags>` arguments).

**Reply by Morganamilo** [reply]
The main reason I never added this was because it would mean every one would have to retranslate the help. If some one can do the change and script all the translation refactoring then that would be great. But thats not on my todo right now.

**Comment by ShuiHuo** [comment]
Add date and time  localization ?

```
paru -Si aur/paru

First Submitted : Mon, 19 Oct 2020 00:43:50 +0000
Last Modified   : Thu, 04 Nov 2021 01:08:53 +0000
```


```
env LC_ALL="zh_CN.UTF-8" TZ='PRC' paru -Si aur/paru

首次提交  : 2020年10月19日 星期一 08:43:50
最后更新  : 2021年11月4日 星期四 09:08:53
```

**Reply by Morganamilo** [reply]
Is your output what paru does or what paru should do?

**Reply by ShuiHuo** [reply]
The top is paru original output, and the bottom is what I'd like to see.

**Reply by Morganamilo** [reply]
Thought so. This would need support in the library we use for time so this is probably unlikely.

**Reply by Morganamilo** [reply]
Actually there's an unstable feature for it so may be possible.

**Comment by Morganamilo** [comment]
Hey all, it's been a while due to changing job and moving house and all that. It's been ~5 months since the last proper release of paru now. Partly from the life stuff and also because the next release is planned to be v2 and there's a lot of stuff to push through.

v2 is starting to finalise now and there shouldn't be any new strings. So now is a good time to update translations. Thanks for all the work so far.

**Reply by Morganamilo** [reply]
Actually this is a lie as I realised I've not written the help descriptions for the new flags yet. Maybe hold off until updates until then :P 

I'll get it done within a week.

**Reply by Morganamilo** [reply]
The help text has been updated.

**Comment by alosarjos** [comment]
Hi there!

I was going to update the spanish translation. But when trying to run the `./scripts/mkpot` script I'm getting a 

```
Error: No such file or directory (os error 2)
```

`xtr` is installed and I can call it from the terminal

**Reply by Morganamilo** [reply]
xtr updated and changed the syntax of the command a bit. Fixed now.

**Comment by Morganamilo** [comment]
Hey every one. There's finally a new release of paru out after way too long. To prevent futher delay to the release, this release has gone out as is and the intention is to pull translation updates in a following .1 release.

Conflicts have been a massive pain both for me and translators too. So now the main template is only updated before release/when it makes sense to. So from now on one PR updating one language shouldn't cause a conflict for PRs on other languages.

This v2.0.0 release is big and has a lot of string changes. So for those still interested in helping out now is a good time to update translations. There shouldn't be massive string changes for a good while now.

With this release out. Paru should be back to the regular bugfix releases and small feature updates every once in a while.

I'm also considering moving to transifex if translators prefer that instead of making PRs.

Thanks to every one who's helped out with localisation. Sorry for the conflict hell and and really slow pace during the last release cycle.

**Reply by juxuanu** [reply]
Moving to any platform for translating makes it much easier from my experience.
